src/lib.rs

```diff
@@ -168,32 +481,106 @@ impl Child {
+    ///
+    /// if let Ok(child) = command.spawn() {
+    ///     match child.id() {
+    ///         Ok(id) => println!("Child's ID is {}", child.id());
+    ///         Err(err) => println!("Cannot get child's ID");
+    ///     } else {
+    ///         println!("notepad didn't start");
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// [get-process-id]: https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-getprocessid
+    pub fn id(&self) -> Result<u32> {
+        // TODO: test this function on Windows
         unsafe {
-            let res = TerminateProcess(self.process_information.hProcess, 0);
+            let process_id = GetProcessId(self.process_information.hProcess);
 
-            if res.as_bool() {
-                Ok(())
-            } else {
-                Err(Error::KillFailed(GetLastError().0))
+            match process_id {
+                0 => Err(Error::GetProcessIdFailed(GetLastError().0)),
+                _ => Ok(process_id),
             }
         }
     }
 }
 
+use windows::Win32::Foundation::CloseHandle;
+
+// This function isn't really necessary, but avoid code repetition
+unsafe fn close_handles(process_info: &PROCESS_INFORMATION) {
+    CloseHandle(process_info.hProcess);
+    CloseHandle(process_info.hThread);
+}
+
+/// Describe the result of a process after it has terminated.
+///
+/// This struct is used to represent the exit status or other termination of a
+/// child process. Child processes are created via the [`Command`] struct and
+/// their exit status is exposed through the [`status`][Command::status]
+/// method, or the [`wait`][Child::wait] method of a [`Child`] process.
 pub struct ExitStatus(u32);
 
 impl ExitStatus {
+    /// Success is defined as a zero exit status.
+    ///
+    /// This function return `true` if the `ExitStatus` is zero and `false`
+    /// otherwise.
     pub fn success(&self) -> bool {
         self.0 == 0
     }
 
+    /// Returns the exit code of the process
     pub fn code(&self) -> u32 {
         self.0
     }
 }
 
-unsafe fn close_handles(process_info: &PROCESS_INFORMATION) {
-    CloseHandle(process_info.hProcess);
-    CloseHandle(process_info.hThread);
+use thiserror::Error;
+
+type Result<T> = std::result::Result<T, Error>;
+
+/// Error type of a failed function on a child process.
+///
+/// Note that this error type is not the same of a non-zero exit status. The
+/// error code linked to the error type is the result of the
+/// [`GetLastError`][get-last-error] function. The variants give some context
+/// to the user but if you want more information about an error code, you can
+/// look at the [`System Error Codes`][system-error-codes].
+///
+/// [get-last-error]: https://docs.microsoft.com/en-us/windows/win32/api/errhandlingapi/nf-errhandlingapi-getlasterror
+/// [system-error-codes]: https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes
+#[derive(Error, Debug)]
+pub enum Error {
```

Comment:
It would be nice to have a method code() that returns the numeric code no matter the variant